### ISSUE TITLE
Pandoc 2.0 breaking all `runtime: shiny` docs

### ISSUE THREAD

**Issue by bborgesr** [rstudio/rmarkdown#1206]
When trying to find a minimal repro for https://github.com/rstudio/flexdashboard/issues/158, I found that all `runtime: shiny` Rmd documents do not play well with Pandoc 2.0. It seems that ANY `runtime: shiny` document is vulnerable and that this bug manifests whenever the author interacts with `input` or `output`. At first, I thought this was a consequence of some weird state my Mac got in after everything I've put it through with Pandoc, but I reproduced the exact same bug in another computer (Microsoft) in a completely different (and much cleaner) environment...

It seems that Pandoc 2.0 is fundamentally not playing well with Shiny. I'm not sure where this problem resides. I only know it's not just `flexdashboard`. @yihui, @jcheng5 

Here is the code (from http://shiny.rstudio.com/articles/interactive-docs.html):

````julia
---
runtime: shiny
output: html_document
---
  
### Here are two Shiny widgets
  
```{r echo = FALSE}
library(shiny)

selectInput("n_breaks", label = "Number of bins:",
  choices = c(10, 20, 35, 50), selected = 20)

sliderInput("bw_adjust", label = "Bandwidth adjustment:",
  min = 0.2, max = 2, value = 1, step = 0.2)
```

### ...that build a histogram.

```{r echo = FALSE}
renderPlot({
  hist(faithful$eruptions, probability = TRUE, 
    breaks = as.numeric(input$n_breaks),
    xlab = "Duration (minutes)", 
    main = "Geyser eruption duration")
  
  dens <- density(faithful$eruptions, adjust = input$bw_adjust)
  lines(dens, col = "blue")
})
```
````

It looks good on Pandoc 1.9. When you change to Pandoc 2.0, you get:

![screen shot 2017-11-18 at 3 33 39 am](https://user-images.githubusercontent.com/6527540/32976583-870eb610-cc11-11e7-8e2e-932d2b00243b.png)

Rmarkdown console output:
```r

[WARNING] This document format requires a nonempty <title> element.
  Please specify either 'title' or 'pagetitle' in the metadata.
  Falling back to 'file602280134b4.utf8'


processing file: pandoc2.Rmd
  |................                                                 |  25%
  ordinary text without R code

  |................................                                 |  50%
label: unnamed-chunk-1 (with options) 
List of 1
 $ echo: logi FALSE

  |.................................................                |  75%
  ordinary text without R code

  |.................................................................| 100%
label: unnamed-chunk-2 (with options) 
List of 1
 $ echo: logi FALSE


output file: /private/var/folders/6b/xmhjngk91hd6d61027q69kz40000gn/T/RtmpDJWrSg/pandoc2.knit.md

/usr/local/bin/pandoc +RTS -K512m -RTS /private/var/folders/6b/xmhjngk91hd6d61027q69kz40000gn/T/RtmpDJWrSg/pandoc2.utf8.md --to html --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash+smart --output /private/var/folders/6b/xmhjngk91hd6d61027q69kz40000gn/T/RtmpDJWrSg/file6022397a3c1f.html --email-obfuscation none --standalone --section-divs --template /Library/Frameworks/R.framework/Versions/3.4/Resources/library/rmarkdown/rmd/h/default.html --no-highlight --variable highlightjs=1 --id-prefix section- --variable 'theme:bootstrap' --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' 
[WARNING] This document format requires a nonempty <title> element.
  Please specify either 'title' or 'pagetitle' in the metadata.
  Falling back to 'pandoc2.utf8'

Output created: /private/var/folders/6b/xmhjngk91hd6d61027q69kz40000gn/T/RtmpDJWrSg/file6022397a3c1f.html
Warning: Error in hist.default: object 'input' not found
Stack trace (innermost first):
    103: hist.default
    102: hist
    101: renderPlot [<text>#2]
     91: <reactive:plotObj>
     80: plotObj
     79: origRenderFunc
     78: output$outdd75afb789149b1a
      3: <Anonymous>
      2: do.call
      1: rmarkdown::run
```

Reverting back to Pandoc 1.9 undoes the damage

**Comment by jcheng5** [comment]
I only skimmed but did you try installing rmarkdown and knitr from master? Yihui just fixed an issue that sounded just like this.

**Comment by bborgesr** [comment]
I have both of both... I tried a LOT of things, but I may have missed something

Here's the full session info:
```r
> devtools::session_info()
Session info -------------------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.4.2 (2017-09-28)
 system   x86_64, darwin15.6.0        
 ui       RStudio (1.1.383)           
 language (EN)                        
 collate  en_US.UTF-8                 
 tz       Europe/Lisbon               
 date     2017-11-18                  

Packages -----------------------------------------------------------------------------------------------------------
 package       * version date       source                            
 backports       1.1.1   2017-09-25 CRAN (R 3.4.2)                    
 base          * 3.4.2   2017-10-04 local                             
 compiler        3.4.2   2017-10-04 local                             
 curl            2.8.1   2017-07-21 cran (@2.8.1)                     
 datasets      * 3.4.2   2017-10-04 local                             
 devtools        1.13.4  2017-11-09 CRAN (R 3.4.2)                    
 digest          0.6.12  2017-01-27 CRAN (R 3.4.0)                    
 evaluate        0.10.1  2017-06-24 CRAN (R 3.4.1)                    
 flexdashboard   0.5.1   2017-11-18 local                             
 git2r           0.18.0  2017-01-01 CRAN (R 3.4.0)                    
 graphics      * 3.4.2   2017-10-04 local                             
 grDevices     * 3.4.2   2017-10-04 local                             
 htmltools       0.3.6   2017-04-28 CRAN (R 3.4.0)                    
 httr            1.3.1   2017-08-20 cran (@1.3.1)                     
 jsonlite        1.5     2017-06-01 CRAN (R 3.4.0)                    
 knitr           1.17.14 2017-11-18 Github (yihui/knitr@2b1786a)      
 magrittr        1.5     2014-11-22 CRAN (R 3.4.0)                    
 memoise         1.1.0   2017-04-21 CRAN (R 3.4.0)                    
 methods       * 3.4.2   2017-10-04 local                             
 R6              2.2.2   2017-06-17 CRAN (R 3.4.0)                    
 Rcpp            0.12.13 2017-09-28 cran (@0.12.13)                   
 rmarkdown       1.8     2017-11-18 Github (rstudio/rmarkdown@6757c4a)
 rprojroot       1.2     2017-01-16 CRAN (R 3.4.0)                    
 rsconnect       0.8.5   2017-08-23 CRAN (R 3.4.1)                    
 stats         * 3.4.2   2017-10-04 local                             
 stringi         1.1.5   2017-04-07 cran (@1.1.5)                     
 stringr         1.2.0   2017-02-18 CRAN (R 3.4.0)                    
 tools           3.4.2   2017-10-04 local                             
 utils         * 3.4.2   2017-10-04 local                             
 withr           2.1.0   2017-11-01 CRAN (R 3.4.2)                    
 yaml            2.1.14  2016-11-12 CRAN (R 3.4.0)    
```

**Comment by bborgesr** [comment]
different versions of shiny didn't matter either

**Comment by yihui** [comment]
I cannot reproduce it with Pandoc 2.0.2.

```r
> devtools::session_info('rmarkdown')
Session info -------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.4.2 (2017-09-28)
 system   x86_64, darwin15.6.0        
 ui       RStudio (1.1.383)           
 language (EN)                        
 collate  en_US.UTF-8                 
 tz       America/Chicago             
 date     2017-11-18                  

Packages -----------------------------------------------------------------------------------
 package   * version date       source        
 backports   1.1.1   2017-09-25 CRAN (R 3.4.2)
 base64enc   0.1-3   2015-07-28 CRAN (R 3.4.0)
 digest      0.6.12  2017-01-27 CRAN (R 3.4.0)
 evaluate    0.10.1  2017-06-24 CRAN (R 3.4.1)
 graphics  * 3.4.2   2017-10-04 local         
 grDevices * 3.4.2   2017-10-04 local         
 highr       0.6     2016-05-09 CRAN (R 3.4.0)
 htmltools   0.3.6   2017-04-28 CRAN (R 3.4.0)
 jsonlite    1.5     2017-06-01 CRAN (R 3.4.0)
 knitr       1.17    2017-08-10 CRAN (R 3.4.1)
 magrittr    1.5     2014-11-22 CRAN (R 3.4.0)
 markdown    0.8     2017-04-20 CRAN (R 3.4.0)
 methods   * 3.4.2   2017-10-04 local         
 mime        0.5     2016-07-07 CRAN (R 3.4.0)
 Rcpp        0.12.13 2017-09-28 CRAN (R 3.4.2)
 rmarkdown   1.8     2017-11-17 CRAN (R 3.4.2)
 rprojroot   1.2     2017-01-16 CRAN (R 3.4.0)
 stats     * 3.4.2   2017-10-04 local         
 stringi     1.1.6   2017-11-17 CRAN (R 3.4.2)
 stringr     1.2.0   2017-02-18 CRAN (R 3.4.0)
 tools       3.4.2   2017-10-04 local         
 utils     * 3.4.2   2017-10-04 local         
 yaml        2.1.14  2016-11-12 CRAN (R 3.4.0)

> rmarkdown::pandoc_version()
[1] ‘2.0.2’
```

rmarkdown 1.8 should have fixed this bug: https://github.com/rstudio/rmarkdown/releases/tag/v1.8

**Comment by bborgesr** [comment]
Can't reproduce it anymore either.

**Comment by maurolepore** [comment]
I'm having a similar problem. 

This passes:

```R
---
title: "Works error"
runtime: shiny
output:
  flexdashboard::flex_dashboard:
    theme: united
    orientation: columns
---
```

But this fails:

```R
---
title: "Works error"
runtime: shiny
output:
  flexdashboard::flex_dashboard:
    theme: united
    orientation: columns
    source_code: embed
---
```

The error is this:

```R
pandoc.exe: C:\Users\LeporeM\AppData\Local\Temp\1\RtmpQfED5E\rmarkdown-str2c801541165f.html: openBinaryFile: does not exist (No such file or directory)

...

Warning: Error in : pandoc document conversion failed with error 1
```

The problem occurs on one of my computers but not on another. Both run windows. Here is the info of the computer where the code fails:

```R
rmarkdown::pandoc_version()
#> [1] '2.1.1'
devtools::session_info()
#> Session info -------------------------------------------------------------
#>  setting  value                       
#>  version  R version 3.4.3 (2017-11-30)
#>  system   x86_64, mingw32             
#>  ui       RTerm                       
#>  language (EN)                        
#>  collate  English_United States.1252  
#>  tz       America/New_York            
#>  date     2018-02-21
#> Packages -----------------------------------------------------------------
#>  package   * version date       source        
#>  backports   1.1.2   2017-12-13 CRAN (R 3.4.3)
#>  base      * 3.4.3   2017-11-30 local         
#>  compiler    3.4.3   2017-11-30 local         
#>  datasets  * 3.4.3   2017-11-30 local         
#>  devtools    1.13.4  2017-11-09 CRAN (R 3.4.3)
#>  digest      0.6.15  2018-01-28 CRAN (R 3.4.3)
#>  evaluate    0.10.1  2017-06-24 CRAN (R 3.4.2)
#>  graphics  * 3.4.3   2017-11-30 local         
#>  grDevices * 3.4.3   2017-11-30 local         
#>  htmltools   0.3.6   2017-04-28 CRAN (R 3.4.2)
#>  knitr       1.20    2018-02-20 CRAN (R 3.4.3)
#>  magrittr    1.5     2014-11-22 CRAN (R 3.4.3)
#>  memoise     1.1.0   2017-04-21 CRAN (R 3.4.2)
#>  methods   * 3.4.3   2017-11-30 local         
#>  Rcpp        0.12.15 2018-01-20 CRAN (R 3.4.3)
#>  rmarkdown   1.8     2017-11-17 CRAN (R 3.4.3)
#>  rprojroot   1.3-2   2018-01-03 CRAN (R 3.4.3)
#>  stats     * 3.4.3   2017-11-30 local         
#>  stringi     1.1.6   2017-11-17 CRAN (R 3.4.2)
#>  stringr     1.3.0   2018-02-19 CRAN (R 3.4.3)
#>  tools       3.4.3   2017-11-30 local         
#>  utils     * 3.4.3   2017-11-30 local         
#>  withr       2.1.1   2017-12-19 CRAN (R 3.4.3)
#>  yaml        2.1.16  2017-12-12 CRAN (R 3.4.3)
```

**Comment by yihui** [comment]
@maurolepore https://github.com/rstudio/rmarkdown/blob/master/.github/ISSUE_TEMPLATE.md

Please do install and test the development version here before you report a bug:

```r
devtools::install_github("rstudio/rmarkdown")
```

**Comment by maurolepore** [comment]
Thanks! After struggling with this for days, installing the development version of __rmarkdown__ fixed my problem. 

I'm very sorry for skipping the issue template. I'll read it next time and I soon I'll add it to my own packages. I've learned a good lesson.
